src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,323 @@
+                    {
+                        return window;
+                    }
+                }
+                return null;
+            });
+
+        internal async Task WaitForDteCommandAvailabilityAsync(string command)
+            => await IntegrationHelper.WaitForResultAsync(() => Dte.Commands.Item(command).IsAvailable, expectedResult: true).ConfigureAwait(continueOnCapturedContext: false);
+
+        protected virtual void Dispose(bool disposing)
+        {
+            _requireNewInstance |= (!disposing);
+            Cleanup();
+        }
+
+        private void CleanupDte()
+        {
+            // DTE can still cause a failure or crash during cleanup
+            try
+            {
+                if (_dte == null)
+                {
+                    return;
+                }
+
+                _dte.Documents.CloseAll(vsSaveChanges.vsSaveChangesNo);
+
+                if (_dte.Solution != null)
+                {
+                    // Save the full path to each project in the solution. This is so we can cleanup any folders after the solution is closed.
+                    var projectFiles = new List<string>();
+
+                    foreach (DteProject project in _dte.Solution.Projects)
+                    {
+                        projectFiles.Add(project.FullName);
+                    }
+
+                    // Save the full path to the solution. This is so we can cleanup any folders after the solution is closed.
+                    var solutionFile = _dte.Solution.FullName;
+
+                    _dte.Solution.Close(SaveFirst: false);
+
+                    foreach (var projectFile in projectFiles)
+                    {
+                        var projectFolder = Path.GetDirectoryName(projectFile);
+                        _requireNewInstance |= IntegrationHelper.TryDeleteDirectoryRecursively(projectFolder);
+                    }
+
+                    if (!string.IsNullOrWhiteSpace(solutionFile))
+                    {
+                        var solutionFolder = Path.GetDirectoryName(solutionFile);
+                        _requireNewInstance |= IntegrationHelper.TryDeleteDirectoryRecursively(solutionFolder);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                IntegrationLog.Current.Warning($"Failed to cleanup the DTE.");
+                IntegrationLog.Current.WriteLine($"\t{e}");
+                _requireNewInstance |= true;
+            }
+        }
+
+        private void CleanupHostProcess()
+        {
+            if (_dte != null)
+            {
+                _dte.Quit();
+                _dte = null;
+            }
+
+            IntegrationHelper.KillProcess(_hostProcess);
+            _hostProcess = null;
+        }
+
+        private void CleanupRemotingService()
+        {
+            try
+            {
+                if ((_dte != null) && (!_dte.Commands.Item(VsStartServiceCommand).IsAvailable))
```

Comment:
You might be able to "simplify" this to `if (_dte?.Commands.Item(VsStartServiceCommand).IsAvailable != true)`, but to be honest it's too early for me to fully think about this so do so at your own risk.